modules/sitemaps/sitemaps.php

```diff
@@ -1,782 +1,2370 @@
-		'changefreq' => 'daily',
-		'priority'   => '1.0'
-	);
-	if ( ! empty( $latest_mod ) ) {
-		$blog_home['lastmod'] = jetpack_w3cdate_from_mysql( $latest_mod );
-		header( 'Last-Modified:' . mysql2date( 'D, d M Y H:i:s', $latest_mod, 0 ) . ' GMT' );
-	}
 	/**
-	 * Filter associative array with data to build <url> node and its descendants for site home.
+	 * Retrieve an array of posts sorted by ID.
 	 *
-	 * @module sitemaps
+	 * More precisely, returns the smallest $num_posts posts
+	 * (measured by ID) which are larger than $from_ID.
 	 *
-	 * @since 3.9.0
+	 * @access private
+	 * @since 4.5.0
 	 *
-	 * @param array $blog_home Data to build parent and children nodes for site home.
+	 * @param int $from_ID Greatest lower bound of retrieved post IDs.
+	 * @param int $num_posts Largest number of posts to retrieve.
+	 *
+	 * @return array The posts.
 	 */
-	$url_node = apply_filters( 'jetpack_sitemap_url_home', $blog_home );
-	jetpack_sitemap_array_to_simplexml( array( 'url' => $url_node ), $tree );
-	unset( $blog_home );
+	private function query_posts_after_ID ( $from_ID, $num_posts ) {
+		global $wpdb;
+
+		$query_string = "
+			SELECT *
+				FROM $wpdb->posts
+				WHERE post_status='publish' AND ID>$from_ID
+				ORDER BY ID ASC
+				LIMIT $num_posts;
+		";
+
+		return $wpdb->get_results( $query_string );
+	}
+
+
 
 	/**
-	 * Filter data before rendering it as XML.
+	 * Retrieve an array of image posts sorted by ID.
 	 *
-	 * @module sitemaps
+	 * More precisely, returns the smallest $num_posts image posts
+	 * (measured by ID) which are larger than $from_ID.
 	 *
-	 * @since 3.9.0
+	 * @access private
+	 * @since 4.5.0
 	 *
-	 * @param SimpleXMLElement $tree Data tree for sitemap.
-	 * @param string $latest_mod Date of last modification.
+	 * @param int $from_ID Greatest lower bound of retrieved image post IDs.
+	 * @param int $num_posts Largest number of image posts to retrieve.
+	 *
+	 * @return array The posts.
 	 */
-	$tree = apply_filters( 'jetpack_print_sitemap', $tree, $latest_mod );
-
-	$xml = $tree->asXML();
-	unset( $tree );
-	if ( ! empty( $xml ) ) {
-		set_transient( 'jetpack_sitemap', $xml, DAY_IN_SECONDS );
-		echo $xml;
+	private function query_images_after_ID ( $from_ID, $num_posts ) {
+		global $wpdb;
+
+		$query_string = "
+			SELECT *
+				FROM $wpdb->posts
+				WHERE post_type='attachment'
+								AND post_mime_type IN ('image/jpeg','image/png','image/gif')
+								AND ID>$from_ID
+				ORDER BY ID ASC
+				LIMIT $num_posts;
+		";
+
+		return $wpdb->get_results( $query_string );
 	}
 
-	die();
-}
-
-/**
- * Prints the news XML sitemap conforming to the Sitemaps.org protocol.
- * Outputs an XML list of up to 1000 posts published in the last 2 days.
- *
- * @module sitemaps
- *
- * @link http://sitemaps.org/protocol.php Sitemaps.org protocol.
- */
-function jetpack_print_news_sitemap() {
 
-	$xml = get_transient( 'jetpack_news_sitemap' );
 
-	if ( $xml ) {
-		header( 'Content-Type: applicati
```

Comment:
Maybe this can be loaded (and any related code executed) only when `WP_DEBUG` is defined and set to `true`?